In [ ]:
# Import the modules
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.metrics import confusion_matrix, classification_report

---

## Split the Data into Training and Testing Sets

### Step 1: Read the `lending_data.csv` data from the `Resources` folder into a Pandas DataFrame.

In [ ]:
# Read the CSV file from the Resources folder into a Pandas DataFrame
lending_data_df = pd.read_csv(
    Path("lending_data.csv")
)

# Review the DataFrame
lending_data_df.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0

### Step 2: Create the labels set (`y`)  from the “loan_status” column, and then create the features (`X`) DataFrame from the remaining columns.

In [ ]:
# Separate the data into labels and features

# Separate the y variable, the labels
y = lending_data_df['loan_status']

# Separate the X variable, the features
X = lending_data_df.drop(columns=['loan_status'])

In [ ]:
# Review the y variable Series
y.head()

0    0
1    0
2    0
3    0
4    0
Name: loan_status, dtype: int64

In [ ]:
# Review the X variable DataFrame
X.head()


loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  
0                 1       22800  
1                 0       13600  
2                 0       16100  
3                 1       22700  
4                 1       23000

### Step 3: Split the data into training and testing datasets by using `train_test_split`.

In [ ]:
# Import the train_test_learn module
from sklearn.model_selection import train_test_split

# Split the data using train_test_split
# Assign a random_state of 1 to the function
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=1)

---

## Create a Logistic Regression Model with the Original Data

###  Step 1: Fit a logistic regression model by using the training data (`X_train` and `y_train`).

In [ ]:
# Import the LogisticRegression module from SKLearn
from sklearn.linear_model import LogisticRegression

# Instantiate the Logistic Regression model
# Assign a random_state parameter of 1 to the model
lr_model = LogisticRegression(random_state=1)
# Fit the model using training data
lr_model.fit(X_train, y_train)

LogisticRegression(random_state=1)

### Step 2: Save the predictions on the testing data labels by using the testing feature data (`X_test`) and the fitted model.

In [ ]:
# Make a prediction using the testing data
y_pred = lr_model.predict(X_test)

### Step 3: Evaluate the model’s performance by doing the following:

* Generate a confusion matrix.

* Print the classification report.

In [ ]:
# Generate a confusion matrix for the model
confusion_matrix(y_test, y_pred)

array([[18655,   110],
       [   36,   583]])

In [ ]:
# Print the classification report for the model
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      0.99      1.00     18765
           1       0.84      0.94      0.89       619

    accuracy                           0.99     19384
   macro avg       0.92      0.97      0.94     19384
weighted avg       0.99      0.99      0.99     19384



### Step 4: Answer the following question.

**Question:** How well does the logistic regression model predict both the `0` (healthy loan) and `1` (high-risk loan) labels?

**Answer:** The model does a great job in using the original data to predicting the values of the healty loans. Precision was 100% and recall was 99%.
The model is pretty good at prediction the values of high risk loans, but not as good as predicting the healty loans. Prediction was 84% and recall was 94%.


---

## Overview of the Analysis


**Purpose of the analysis**

 The purpose of this analysis is to assess whether the Logistic Regression machine learning model can more accurately predict healthy loans versus high-risk loans, using either the original dataset or a resampled dataset designed to increase the size of the minority class.

* **The Dataset** -The dataset used for this analysis contains information on 77,536 loans. The data includes columns for loan_size, interest_rate, borrower_income, debt_to_income ratio, number_of_accounts, derogatory_marks, total_debt, and loans_status. The category that we are attemting to predict with our analysis is **loan_status**. The data provided in the remaining columns will be used as features to train the data and inform the predictions.

* **Stages of the Machine Learning Process**  -The stages of the machine learning process are very scripted. If followed in order, they provide you with an accurate assessment of model's ability to make predictions. The stages of the machine learning process are as follows:

  - Prepare the Data: Import the file, establish the DataFrame, evaluate the columns and features.

  - Separate the Data into feature and labels: The labels represent the target variable
you're trying tp predict-whether the loan status is healthy(0) or high-risk(1). The feautures include all other data that will be used to train and test the model.

  - Split the Data: Use the train_test_split fuction to separate the feautures and labels data into training and testing datasets.

  - Import the Model: Import the machine learning model from the appropriate library, we will be importing logisticRegression model from scikit-learn (SKLearn).

  - Instantiate the model.

  - Fit the model using the training data.

  - Use the model to make predictions using the features test data.

  - Evaluate the predictions: Evaluation is performed by calculating and comparing metrics such as accuracy score, confusion matrix, and classification report.

* **Machine Learning Methods Utillized**  

  The primary model used in this analysis is:

    - LogisticRegression model from SKLearn

  Supporting functions used in this analysis are:

    - train_test_split from SKLearn

  Models are evaluated using the following functions:

  - confusion_matrix from SKLearn
  - classification_report from SKLearn

## Results

Using bulleted lists, describe the accuracy scores and the precision and recall scores of all machine learning models.

* Machine Learning Model-Logistic Regression:
 * **Accuracy Score:** 99%
 * **Precision Scores:**
   * Class 0 (Healthy Loans): 100%
   * Class 1 (High-risk Loans): 84%
 * **Recall Scores:**
   * Class 0 (Healthy Loans): 99%
   * Class 1 (High-risk Loans): 94%

## Summary

The model does a great job in using the original data to predicting the values of the healty loans. Precision was 100% and recall was 99%.

The model is pretty good at prediction the values of high risk loans, but not as good as predicting the healty loans. Prediction was 84% and recall was 94%.

Given this information, it appears that the logic Regression model does a great job at predicting both healthy and high-risk loans, given the feautures that are used to train the data.


